spyder/plugins/ipythonconsole/comms/kernelcomm.py

```diff
@@ -62,12 +65,27 @@ def shutdown_comm_channel(self):
             channel.send(msg)
             self.kernel_client.comm_channel = None
 
+    def comm_channel_connected(self):
+        """Check if the comm channel is connected."""
+        return self.kernel_client.comm_channel is not None
+
     @contextmanager
-    def comm_channel_manager(self, comm_id):
+    def comm_channel_manager(self, comm_id, queue_message=False):
         """Use comm_channel instead of shell_channel."""
-        if not self.kernel_client.comm_channel:
+        if queue_message:
+            # Send without comm_channel
             yield
             return
+
+        if not self.comm_channel_connected():
+            # Ask again for comm config
+            self.remote_call()._send_comm_config()
+            timeout = 10
```

Comment:
Is this 10 seconds? If that's the case, why do it need to be that big?

Reply:
I have to wait for the kernel to launch and connect, which might take a while. 1s is too short. Also this is something that is only waited on once.

Reply:
Ok, then I have another question: what happens if the kernel takes more than 10 secs to launch? That's not an atypical case on Windows (I've seen cases where it takes 30 or 40 secs, and it could take even more after we merge PR #11327).

Reply:
Ok, then I have another question: what happens if the kernel takes more than 10 secs to launch? That's not an atypical case on Windows (I've seen cases where it takes 30 or 40 secs, and it could take even more after we merge PR #11327).

Reply:
Then the function that is making the call would fail.

Reply:
But the next function would wait again

Reply:
I tried adding `time.sleep(15)` in `start.py::main()` and nothing bad happened.